Extension/src/LanguageServer/editorConfig.ts

```diff
@@ -61,13 +61,408 @@ export function mapWrapToEditorConfig(value: string | undefined): string {
     return "never";
 }
 
-function matchesSection(filePath: string, section: string): boolean {
-    const fileName: string = path.basename(filePath);
-    // Escape all regex special characters except '*' and '?'.
-    // Convert wildcards '*' to '.*' and '?' to '.'.
-    const sectionPattern = section.replace(/[.+^${}()|[\]\\]/g, '\\$&').replace(/\*/g, '.*').replace(/\?/g, '.');
-    const regex: RegExp = new RegExp(`^${sectionPattern}$`);
-    return regex.test(fileName);
+let isValid: boolean = true;
+let relativeToCurrentDir: boolean = false;
+
+// Helper function to find matching '}' for a given '{' position
+function findMatchingBrace(pattern: string, start: number): number {
+    let braceLevel = 0;
+    let i = start;
+    while (i < pattern.length) {
```

Comment:
I think it would simplify the code to use

```TypeScript
for (let i = start; i < pattern.length; i++) {
```
and change the 

```
i += 2;
```
to 

```
i++;
```